qualtran/surface_code/thc_compliation.ipynb

```diff
@@ -0,0 +1,189 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## THC FeMoco compliation\n",
+    "\n",
+    "This notebook is attempting to reproduce results from sec. IV.C of (10.1103/PRXQuantum.2.030305)[https://link.aps.org/doi/10.1103/PRXQuantum.2.030305] (here on \"THC paper\"), regarding surface code compliation of quantum phase estimation on the THC-encoded Li et al. FeMoco Hamiltonian."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from functools import partial\n",
+    "\n",
+    "from qualtran.surface_code.algorithm_summary import AlgorithmSummary\n",
+    "from qualtran.surface_code.multi_factory import MultiFactory\n",
+    "from qualtran.surface_code.data_block import SimpleDataBlock\n",
+    "from qualtran.surface_code.ccz2t_cost_model import get_ccz2t_costs_from_grid_search, \\\n",
+    "                                                   iter_ccz2t_factories, \\\n",
+    "                                                   get_ccz2t_costs, \\\n",
+    "                                                   CCZ2TFactory\n",
+    "\n",
+    "# Logical data (input for grid search), from THC paper\n",
+    "ERR_BUDGET = 0.01\n",
+    "N_TOFFOLI = 6665400000 # Pag. 26\n",
+    "N_QUBITS = 696 # Fig. 10 \n",
+    "PHYS_ERR = 1e-3\n",
+    "N_FACTORIES = 4\n",
+    "CYCLE_TIME = 1\n",
+    "\n",
+    "n_magic = AlgorithmSummary(toffoli_gates=N_TOFFOLI)\n",
+    "\n",
+    "\n",
+    "# Manual optimal values (output of grid search) from THC paper\n",
+    "DISTILLATION_L1_D = 19\n",
+    "DISTILLATION_L2_D = 31\n",
+    "DATA_D = 31\n",
+    "FOOTPRINT = int(4E10)\n",
+    "DURATION_DAYS = 3\n",
+    "\n",
+    "target_factory = MultiFactory(CCZ2TFactory(DISTILLATION_L1_D, DISTILLATION_L2_D), N_FACTORIES)\n",
+    "target_data_block = SimpleDataBlock(DATA_D)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Using manual values for factory and data block"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "distillation error: 0.069%\n",
+      "data error: 0.373%\n",
+      "failure probability: 0.442%\n",
+      "wall time: 4.1369453125 days\n",
+      "footprint: 2.81 million qubits\n"
+     ]
```

Comment:
before we merge, make sure you do `All Cells -> Clear Output` from juypter notebook. We don't commit the notebooks to reduce churn in the diffs